crates/types/src/data.rs

```diff
@@ -526,6 +527,45 @@ impl<TYPES: NodeType> Leaf<TYPES> {
     pub fn get_payload_commitment(&self) -> VidCommitment {
         self.get_block_header().payload_commitment()
     }
+
+    /// Validate that a leaf is the immediate child of another leaf
+    ///
+    /// This may not be a complete function. Please double-check that it performs the checks you expect before subtituting validation logic with it.
+    pub fn extends(
```

Comment:
Nit: can we rename this to extends_upgrade or something.  I think we're going to want an `extends` on leaf at some point to replace the safety check